pkg/kubeapiserver/authorizer/config.go

```diff
@@ -98,6 +98,7 @@ func (config Config) New(ctx context.Context, serverID string) (authorizer.Autho
 				config.VersionedInformerFactory.Core().V1().Pods(),
 				config.VersionedInformerFactory.Core().V1().PersistentVolumes(),
 				config.VersionedInformerFactory.Storage().V1().VolumeAttachments(),
+				config.VersionedInformerFactory.Resource().V1alpha2().NodeResourceSlices(),
```

Comment:
`AddGraphEventHandlers` ignores the slices informer if the feature is disabled, so this is covered. I already found this while testing because the [`go cache.WaitForNamedCacheSync("node_authorizer", <...>)` goroutine](https://github.com/kubernetes/kubernetes/blob/dc3f5ec6ccb9855dfa99f4c1078625df5fdfab6a/plugin/pkg/auth/authorizer/node/graph_populator.go#L65-L66) got stuck waiting for the NodeResourcesSlice informer.

 I can move the feature gate check here if that's clearer, but then we need two if checks (one here for the feature, one in `AddGraphEventHandlers` for nil) and risk forgetting to remove that second if check when promoting the feature.

Speaking of that `WaitForNamedCacheSync`: does that have some purpose that I don't see? It waits in a goroutine and nothing checks for the result of that `WaitForNamedCacheSync`.

Reply:
`config.VersionedInformerFactory.Resource().V1alpha2().NodeResourceSlices()` starts the informer and will spam the apiserver with 404 requests endlessly. We don't want that in default clusters.

> Speaking of that WaitForNamedCacheSync: does that have some purpose that I don't see? It waits in a goroutine and nothing checks for the result of that WaitForNamedCacheSync.

I had to dig ... looks like it was added in https://github.com/kubernetes/kubernetes/commit/4d81f7e1299b834839dcfea84920718304406a80#diff-a7fa45064914e8de86acedee85a63197be7690107545e0b4aa9524f544084521R81 to push info into the logs when the sync was complete (to help explain rejections by the node authorizer while the cache was filling during server startup).

Reply:
Hmm, right. What "solved" my pending goroutine issue was skipping the sync check, but the informer still ran and failed.

Fixed.